### PR TITLE
Report illegal write for Unsafe.put

### PR DESCRIPTION
**Pull request by liqunl**
Add a diamond to take the original call if the field is static final.

Signed-off-by: Liqun Liu <liqunl@ca.ibm.com>

### CODE REVIEW

**Comment by liqunl** [comment]
@0dvictor Can you help the review?
Depends https://github.com/eclipse/omr/pull/2959

**Comment by 0dvictor** [comment]
LGTM.

**Comment by andrewcraik** [comment]
My main comment is that we need a way to say whether we are needing to report illegal writes - if we never make assumptions on a given class or if we want to disable the opt we need a way to turn that off. Perhaps we need a function somewhere sensible to say are we reporting writes on a given field and perhaps pass the node that is the object so that if we need to be selective in the future we can be.

**Comment by liqunl** [comment]
I have the env var in this commit already. It's `TR_DisableIllegalWriteReport`. This change is ready for a jenkins build.

**Comment by liqunl** [comment]
This PR is part of https://github.com/eclipse/openj9/issues/3018

**Comment by andrewcraik** [comment]
Jenkins test sanity xlinux,win jdk8,jdk11

**Comment by liqunl** [comment]
Perf measured on daytrader3 and Hibench, on regression.

File: runtime/compiler/optimizer/InlinerTempForJ9.cpp [hunk 1]
```diff
@@ -1060,91 +1061,62 @@ TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSy
...
-                              offset, newSymbolReferenceForAddress, false);
+   static char *disableIllegalWriteReport = feGetEnv("TR_DisableIllegalWriteReport");
+   TR::TreeTop* callTreeForIllegalWriteReport = NULL;
+   // Keep the call for illegal write report
+   if (!disableIllegalWriteReport && !comp()->getOption(TR_DisableGuardedStaticFinalFieldFolding))
+      callTreeForIllegalWriteReport = callNodeTreeTop->duplicateTree();
 
    if(comp()->getOption(TR_TraceUnsafeInlining))
        {
        traceMsg(comp(),"\t After createTempsForUnsafePutGet, callNodeTreeTop dump:\n");
```
Comment (andrewcraik) [inline runtime/compiler/optimizer/InlinerTempForJ9.cpp]: since you removed the createTempsForUnsafePutGet this trace is misleading now.

**Comment by andrewcraik** [comment]
Sanity was previously clean and the last update was a simple removal. With a clean CI this is good to go.

**Pull request merged**